src/client/components/multi-form-editor/landing-page.js

```diff
@@ -1,36 +1,224 @@
 class LandingPage extends React.Component {
+  constructor( props ){
+    super( props );
+
+    this.state = {
+      textAreaEnabled: false,
+      submitting: false,
+      textAreaAnimating: false,
+      textAreaHeight: '20em'
+    };
+  }
+
+  createDoc(){
+    let text = ReactDom.findDOMNode(this).querySelector('.landing-page-text').value;
+
+    let makeRequest = () => fetch('/api/document', {
+      headers: {
+        'Content-Type': 'application/json'
+      },
+      method: 'POST',
+      body: JSON.stringify({ text })
+    });
+
+    let toJson = res => res.json();
+
+    let animateResult = () => {
+      let linkout = ReactDom.findDOMNode(this).querySelector('.landing-page-linkout');
+
+      if( linkout ){
+        anime({ targets: linkout, opacity: [0, 1], duration: 2000, easing: 'linear' });
+      }
+    };
+
+    let updateState = documentJson => {
+      documentJson.editable = true;
+
+      this.setState({ documentJson, submitting: false }, animateResult);
+    };
+
+    this.setState({ submitting: true });
+
+    Promise.try( makeRequest ).then( toJson ).then( updateState );
+  }
+
   render(){
     return h('div.landing-page', [
-      // main part of landing-page page
-      h('div.landing-page-center', [
-        // A row that includes Factoid logo and short description of Factoid
+      h('div.landing-page-center', {
+        className: makeClassList({
+          'landing-page-center-fit': !this.state.textAreaEnabled
+        })
+      },[
         h('div.landing-page-intro-row', [
           h('i.landing-page-icon'),
           h('div', [
             h('h1.landing-page-into-header', 'Factoid'),
             h('p.landing-page-intro-desc', 'A project to digitally capture biological data from academic papers')
           ])
         ]),
-        // Links to choose the mode
-        h('div.landing-page-mode-links', [
-          h('span.landing-page-mode-link', [ h(Link, { className: 'plain-link', to: '/debug/new-document' }, 'Create new, blank document') ]),
-          h('span.landing-page-mode-link-seperator', '-'),
-          h('span.landing-page-mode-link', [ h(Link, { className: 'plain-link', to: '/debug/new-document/fill' }, 'Create new document, filled from text') ])
-        ])
+        h('div.landing-page-mode-area', [
+          h('span', {
+            className: makeClassList({
+              'landing-page-hidden': this.state.textAreaEnabled
+            })
+          }, 'CREATE'),
+          h('button.landing-page-fill-text-btn', {
+            className: makeClassList({
+              'landing-page-left': this.state.textAreaEnabled,
+              'landing-page-no-right-border': !this.state.textAreaEnabled,
+              'landing-page-no-bottom-border': this.state.textAreaEnabled,
+              'landing-page-round-tl': true,
+              'landing-page-round-bl': !this.state.textAreaEnabled,
+              'landing-page-round-tr': this.state.textAreaEnabled
+            }),
+            onClick: () => {
```

Comment:
this is quite a big function. I'd rather have it at class level (a.k.a.: at the same level of `render`). And then call it here via `onClick={this.something}`.